webxrlayers-1.bs

```diff
@@ -385,6 +385,11 @@ to this layer when {{XRCompositionLayer/needsRedraw}} is <code>true</code>.
 NOTE: if [=XRCompositionLayer/isStatic=] is <code>true</code> the author can only draw into the layer once after creation or once after
 a redraw event. This allows the UA to only allocate a single GPU buffer.
 
+When a writeable attribute is set on an {{XRCompositionLayer}} or any of its derived classes, reading that
+attribute MUST return that value.
+The value MUST also immediately be sent to the underlying [=XR Compositor=]. The [=XR Compositor=] MUST apply
+the value at its earliest convenience.
```

Comment:
I want to make sure I understand what this is proposing: Sounds to me like any time I change a writable attribute on a layer (ie: `width` or `height` on an `XRQuadLayer`) that change must immediately be communicated to the XR Compositor, regardless of when in the frame loop it was set, and the XR Compositor must start using it at the first possible moment.

This seems like it offers an opportunity for race conditions or unpredictable behavior. For instance, if I have a quad layer that I render to, allow the frame callback to end, then adjust the width and height for the next frame it seems plausible that the width/height change may make it back to the compositor before the frame's rendered results do, resulting in them take effect a frame earlier than I expected.

My intuition would be that we would want to effectively capture the layer state after the frame callbacks have completed and guarantee that the composited content will be displayed with those attributes, with any subsequent changes not taking effect till the next frame?

Reply:
OK. Maybe I can update the spec so it states that any attributes if changed, must be sent to the XR Compositor at the end of the frame and used when that frame is displayed.
What do you think?

Reply:
That would feel far more predictable to me, yes. Thanks!

Reply:
+1